### PR TITLE
Feature/issue 238

### LINKED ISSUE #238: Query status "whether an address has been paid"
**Is your feature request related to a problem? Please describe.**
I want to know if an address has been paid

**Describe the solution you'd like**
I want to query the payment status of the corresponding NFT by class_id, nft_id, bidder address.
- add parameter to bid?
- Increase dedicated queries?

### PR DESCRIPTION
**Pull request by mkXultra**
fix #238

### CODE REVIEW

File: docs/core/proto-docs.md [hunk 1]
```diff
@@ -1943,6 +1946,28 @@ GenesisState defines the nftmarket module's genesis state.
...
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| `nft_id` | [NftIdentifier](#ununifi.nftmarket.NftIdentifier) |  |  |
+| `bidder` | [string](#string) |  |  |
+| `amount` | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) |  |  |
+| `automatic_payment` | [bool](#bool) |  |  |
+| `paid_amount` | [string](#string) |  |  |
+| `bid_time` | [google.protobuf.Timestamp](#google.protobuf.Timestamp) |  |  |
+| `state` | [ListingState](#ununifi.nftmarket.ListingState) |  |  |
+| `all_paid` | [bool](#bool) |  |  |
```
Comment (jununifi) [inline docs/core/proto-docs.md:1964]: Can the table more formatted? There's automatic formatter extention on VScode

File: docs/core/proto-docs.md [hunk 2]
```diff
@@ -2076,40 +2101,46 @@ GenesisState defines the nftmarket module's genesis state.
...
+<a name="ununifi.nftmarket.QueryListedNftsResponse"></a>
 
-### QueryListedNftsRequestM
+### QueryListedNftsResponse
 
 
 
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| `listings` | [NftListing](#ununifi.nftmarket.NftListing) | repeated |  |
```
Comment (jununifi) [inline docs/core/proto-docs.md:2112]: same here

File: docs/core/proto-docs.md [hunk 3]
```diff
@@ -2228,6 +2259,38 @@ GenesisState defines the nftmarket module's genesis state.
...
+
+### QueryPaymentStatusRequest
+
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| `class_id` | [string](#string) |  |  |
+| `nft_id` | [string](#string) |  |  |
+| `bidder` | [string](#string) |  |  |
```
Comment (jununifi) [inline docs/core/proto-docs.md:2272]: Can the table more formatted? There's automatic formatter extention on VScode

File: docs/core/proto-docs.md [hunk 4]
```diff
@@ -2228,6 +2259,38 @@ GenesisState defines the nftmarket module's genesis state.
...
+
+<a name="ununifi.nftmarket.QueryPaymentStatusResponse"></a>
+
+### QueryPaymentStatusResponse
+
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| `paymentStatus` | [PaymentStatus](#ununifi.nftmarket.PaymentStatus) |  |  |
```
Comment (jununifi) [inline docs/core/proto-docs.md:2287]: Can the table more formatted? There's automatic formatter extention on VScode

File: x/nftmarket/keeper/grpc_query_test.go [hunk 5]
```diff
@@ -219,3 +219,80 @@ func (s *KeeperTestSuite) TestListedNfts() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestLoan() {
+	// var req *types.QueryLoanRequest
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query_test.go:224]: Can you remove commented code?

File: x/nftmarket/keeper/grpc_query_test.go [hunk 6]
```diff
@@ -219,3 +219,80 @@ func (s *KeeperTestSuite) TestListedNfts() {
...
+		msg       string
+		malleate  func(index int, require *require.Assertions)
+		req       *types.QueryLoanRequest
+		expError  string
+		expResult types.Loan
+	}{
+		{
+			"success empty",
+			func(index int, require *require.Assertions) {
+				// req = &types.QueryListedNftsRequest{}
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query_test.go:235]: Can you remove commented code?

File: docs/core/proto-docs.md (hunk 1 above)
Comment (mkXultra) [reply docs/core/proto-docs.md:1964]: Which extension of vscode do you use?

File: x/nftmarket/keeper/grpc_query_test.go (hunk 5 above)
Comment (mkXultra) [reply x/nftmarket/keeper/grpc_query_test.go]: fixed

File: x/nftmarket/keeper/grpc_query_test.go (hunk 6 above)
Comment (mkXultra) [reply x/nftmarket/keeper/grpc_query_test.go]: fixed

File: x/nftmarket/keeper/grpc_query.go [hunk 7]
```diff
@@ -209,3 +224,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
 	return &types.QueryRewardsResponse{}, nil
 }
+
+func (k Keeper) PaymentStatus(c context.Context, req *types.QueryPaymentStatusRequest) (*types.QueryPaymentStatusResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	_ = ctx
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:234]: This line could be removed since it's not used

File: x/nftmarket/keeper/grpc_query.go [hunk 8]
```diff
@@ -209,3 +224,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
+		return nil, status.Error(codes.InvalidArgument, "not existing bidder")
+	}
+
+	var bidderBid types.NftBid
+	for _, v := range bids {
+		if v.Bidder == req.Bidder {
+			bidderBid = v
+		}
+	}
+	if (bidderBid == types.NftBid{}) {
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:254]: `()` could be removed on if clause

File: x/nftmarket/keeper/grpc_query.go [hunk 9]
```diff
@@ -209,3 +224,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
+		return nil, status.Error(codes.InvalidArgument, "does not match bidder")
+	}
+
+	allPaid := listing.State >= types.ListingState_END_LISTING && bidderBid.Amount.Amount.Equal(bidderBid.PaidAmount)
+	fmt.Println(allPaid)
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:259]: log function could be removed

File: x/nftmarket/keeper/grpc_query.go [hunk 10]
```diff
@@ -209,3 +224,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
+	return &types.QueryPaymentStatusResponse{
+		PaymentStatus: types.PaymentStatus{
+			NftId:            listing.NftId,
+			State:            listing.State,
+			Bidder:           bidderBid.Bidder,
+			Amount:           bidderBid.Amount,
+			AutomaticPayment: bidderBid.AutomaticPayment,
+			PaidAmount:       bidderBid.PaidAmount,
+			BidTime:          bidderBid.BidTime,
+			AllPaid:          false,
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:269]: allPaid is hardcoded as false - shouldn't it be `AllPaid: allPaid` ?

**Pull request merged**